components/Modals/JoinGameModal.tsx

```diff
@@ -0,0 +1,416 @@
+            </FormHelperText>
+          )}
+        </FormControl>
+        <FormControl isInvalid={showError && !characterDescription}>
+          <FormLabel>Character Description (200 character limit)</FormLabel>
+          <Textarea
+            onChange={e => setCharacterDescription(e.target.value)}
+            value={characterDescription}
+          />
+          {showError && !characterDescription && (
+            <FormHelperText color="red">
+              A character description is required
+            </FormHelperText>
+          )}
+          {showError && invalidCharacterDescription && (
+            <FormHelperText color="red">
+              Character description must be less than 200 characters
+            </FormHelperText>
+          )}
+        </FormControl>
+        <FormControl isInvalid={showError && !characterAvatar}>
+          <FormLabel>Character Avatar</FormLabel>
+          {!characterAvatar && (
+            <Input
+              accept=".png, .jpg, .jpeg, .svg"
+              disabled={isUploading}
+              onChange={e => setCharacterAvatar(e.target.files?.[0] ?? null)}
+              type="file"
+              variant="file"
+            />
+          )}
+          {characterAvatar && (
+            <Flex align="center" gap={10} mt={4}>
+              <Image
+                alt="character avatar"
+                objectFit="contain"
+                src={URL.createObjectURL(characterAvatar)}
+                w="300px"
+              />
+              <Button
+                isDisabled={isUploading || isUploaded}
+                isLoading={isUploading}
+                loadingText="Uploading..."
+                mt={4}
+                onClick={!isUploaded ? onRemove : undefined}
+                type="button"
+                variant="outline"
+              >
+                {isUploaded ? 'Uploaded' : 'Remove'}
+              </Button>
+            </Flex>
+          )}
+          {showError && !characterAvatar && (
+            <FormHelperText color="red">
+              A character avatar is required
+            </FormHelperText>
+          )}
+        </FormControl>
+
+        <Button
+          alignSelf="flex-end"
+          isDisabled={isDisabled}
+          isLoading={isLoading}
+          loadingText="Creating..."
+          type="submit"
+        >
+          Create
+        </Button>
+      </VStack>
+    );
+  };
+
+  return (
+    <Modal
+      closeOnEsc={!isLoading}
+      closeOnOverlayClick={!isLoading}
+      isOpen={isOpen}
+      onClose={onClose}
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>
+          <Text>Create your Character</Text>
+          <ModalCloseButton size="lg" />
+        </ModalHeader>
+        <ModalBody>{content()}</ModalBody>
+      </ModalContent>
+    </Modal>
+  );
+};
+
+type TransactionPendingProps = {
+  isSyncing: boolean;
+  txHash: string;
+};
+
+const TransactionPending: React.FC<TransactionPendingProps> = ({
```

Comment:
May just move this to its own component, since it's also used in `CreateGameModal`